frontend/packages/ceph-storage-plugin/src/components/modals/block-pool-modal/modal-footer.tsx

```diff
@@ -42,57 +43,118 @@ export const BlockPoolModalFooter = (props: BlockPoolModalFooterProps) => {
     close();
   };
 
-  return state.poolStatus ? (
+  return (
     <ActionGroup className="pf-c-form pf-c-form__actions--right pf-c-form__group--no-top-margin">
-      {state.poolStatus === POOL_PROGRESS.FAILED && (
-        <Button
-          type="button"
-          variant="secondary"
-          data-test="modal-try-again-action"
-          onClick={handleTryAgainButton}
-        >
-          {t('ceph-storage-plugin~Try Again')}
-        </Button>
+      {state.poolStatus ? (
+        // Action wise status check
+        actionButtonTextMapping === ActionButtonTextMapping.CREATE ? (
```

Comment:
I modified footer logic to remove hacky logic, old logic is tightly coupled and very difficult to do any changes,
Now all buttons are specific to its main action and different pool status. 

any changes in the future can easily doable

Reply:
logic is 
{
  main_action_name: {
     pool status ; [buttons],
     .
     .
     .
  }
  .
  .
  .
}